meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1634 @@
+            context.scene.frame_end = num_keyframes
+
+        joint_names = get_joint_names(SMPL_version)
+        joints_to_use = joint_names
+
+        # override hand pose if it's selected
+        # don't pose the hands every frame if we're overriding it
+        if (self.hand_pose != 'disabled'):
+            bpy.ops.object.set_hand_pose('EXEC_DEFAULT')
+
+            if SMPL_version == 'SMPLH':
+                joints_to_use = joint_names[:22]
+            else:
+                joints_to_use = joint_names[:25]
+                print (joints_to_use)
+
+        for index, frame in enumerate(range(0, num_frames, step_size)):
+            if (index % 100) == 0:
+                print(f"  {index}/{num_keyframes}")
+            current_frame = index + 1
+            current_pose = poses[frame].reshape(-1, 3)
+            current_trans = trans[frame]
+
+            for bone_index, bone_name in enumerate(joints_to_use):
+                if bone_name == "pelvis":
+                    # there's a scale mismatch somewhere and the global translation is off by a factor of 100
+                    armature.pose.bones[bone_name].location = Vector((current_trans[0]*100, current_trans[1]*100, current_trans[2]*100))
+                    armature.pose.bones[bone_name].keyframe_insert('location', frame=current_frame)
+
+                # Keyframe bone rotation
+                pose_rodrigues = current_pose[bone_index]
+                set_pose_from_rodrigues(armature, bone_name, pose_rodrigues)
+                armature.pose.bones[bone_name].keyframe_insert('rotation_quaternion', frame=current_frame)
+
+            if self.keyframe_corrective_pose_weights:
+                # Calculate corrective poseshape weights for current pose and keyframe them.
+                # Note: This significantly increases animation load time and also reduces real-time playback speed in Blender viewport.
+                bpy.ops.object.update_pose_correctives('EXEC_DEFAULT')
+                for key_block in obj.data.shape_keys.key_blocks:
+                    if key_block.name.startswith("Pose"):
+                        key_block.keyframe_insert("value", frame=current_frame)
+
+        correct_for_anim_format(self.anim_format, armature)
+
+        print(f"  {num_keyframes}/{num_keyframes}")
+        context.scene.frame_set(1)
+
+        return {'FINISHED'}
+
+
+class OP_CreateAvatar(bpy.types.Operator):
+    bl_idname = "scene.create_avatar"
+    bl_label = "Create Avatar"
+    bl_description = ("Create a SMPL family avatar at the scene origin.  \nnote: SMPLH is missing the joint regressor so you can't modify it's shape")
+    bl_options = {'REGISTER', 'UNDO'}
+
+    @classmethod
+    def poll(cls, context):
+        try:
+            # Enable button only if in Object Mode
+            SMPL_version = context.window_manager.smpl_tool.SMPL_version
+
+            if (context.active_object is None) or (context.active_object.mode == 'OBJECT'):
```

Comment:
You can always just say `return some_condition` instead of `if some_condition return true else return false`